### DISCUSSION TITLE
Ability to use multiple profiles in one window via workspaces

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#818] [Ideas]
### What feature would you like to see?

One of the killer features for me in the Arc Browser is the ability to associate a profile with the workspace, which allows to isolate workspaces from each other and have different sets of configurations, extensions, history, favorites, cookies, etc in every workspace. Unlike vanilla Chrome or Firefox, switching between workspaces with different profiles happens without creating a new window, seamlessly and transparently, which allows me to not even care about extra window management.

I have 3 different Microsoft and Google accounts which I use for different purposes (two for work and one for own business), and I have 3 different workspaces, each one has exactly one logged in MS account. I also have different sets of extensions necessary for every workspace, and in some cases I have same extensions with different configurations, e.g. Bitwarden that uses different accounts based on the workspace I am currently in.

Currently in Zen browser I can use workspaces only as an organizational feature, where every workspace is like a container with different sets of tabs, but the sessions, cookies, extensions and browser preferences, etc. are shared between all of them.

I would like to have an ability to associate a workspace with different profile in the same way as it is done in Arc, to be able to use my workspaces as fully isolated (or not, if there are sibling workspaces associated with the same profile) environments and have a better browsing experience, and switch these workspaces in the same browser window.

### What platform are you seeing the problem on?

Windows

 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/500_

**Comment by dshan12** [comment]
This is the only thing that is stopping me from switching from arc to the zen browser. The second this is implemented I will definitely switch.

**Reply by japboy** [reply]
Firefox containers are incomplete-sandbox. It shares search histories etc between different containers. There is a unfinished discussion on Github ([here](https://github.com/mozilla/multi-account-containers/issues/952) and [there](https://github.com/mozilla/multi-account-containers/issues/1233)) and [Firefox Bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1283320) ~(I forgot the address though)~ and they are untouched for years. I personally feel Firefox won't solve the issues about the containers anymore.

So, I personally think the workspaces functionality which is the abstraction of the containers might not a good idea for long-time use. And, yes, the profiles may be a good abstraction for the workspaces.

**Reply by Malagutte** [reply]
I feel the same way, I want to get out of Chromium-based browsers

**Reply by stanimir93** [reply]
I need this feature too, and it is is also stopping me from using Zen unfortunately. I hope there is a way to resolve it.

**Comment by judehunter** [comment]
This is such a killer feature in Arc. So killer that it's the only thing preventing me from switching @mauro-balades

**Comment by thestrike72** [comment]
Completely agree. This is what I've searched for in every browser I've tried on linux.

**Reply by manojm-dev** [reply]
firefox has it by default and you can use simple tab group extension to create tab groups and assign it any specific containers, temp container or we can leave it to default.

**Comment by manojm-dev** [comment]
Here is my take/use-case that would require about feature to be implemented

**What’s the Idea?**

In Firefox, there’s this feature called Multi-Account Containers that lets you keep different browsing activities separate. You can have containers for things like shopping, work, social media, etc., which helps with privacy and organization. They also have a neat extension called Simple Tab Groups that lets you group tabs and assign specific containers to these groups.

**Here’s my suggestion:**

What if Zen Browser had something similar? Imagine if we could:

- Create Workspaces with Containers: Like, have a “Shopping” workspace that uses a “Shopping” container, or a “Work” workspace that uses a “Work” container. This way, you can easily switch between different tasks with all your tabs and sessions neatly organized.

- Better Privacy and Organization: This could keep things separated and tidy, making it easier to manage different activities and keeping your data safe.

**Why It’d Be Awesome**

Organized Browsing: Keep different tasks and tabs neatly separated.
Enhanced Privacy: Prevents cross-session tracking and keeps your data isolated.
Increased Productivity: Easy to switch between different tasks without losing context.

What do you all think? Would this be a cool feature to add? Let’s discuss!

**Reply by thestrike72** [reply]
This is a great idea. The biggest problem with this solution I see is if you have to do extra things like enable the multi-account container tabs extensions to get it to work.

My current solution is to manually add things to container tabs that are already in workspaces, or set some sites to always open in certain containers. This requires a lot of work when it really should be 1 click to enable workspaces and have them include the features of multi-account containers.

**Reply by suamai** [reply]
This would be fantastic, if possible!

I'm currently using Multi-Account Containers to separate my work session from my personal one, and also trying to have a Workspace for each - but, without being able to link a workspace to a container, keeping them organized is kinda awkward.

**Comment by jean** [comment]
I added the Multi-Account Containers to Zen. It seems to be working ("Always open this in a Container" shows up in the location bar, but the "New tab +" button does not prompt for a container the way it does in Firefox.) 

You can open container tabs via the extension's menu. 
![image](https://github.com/user-attachments/assets/2c8d80ab-ff72-4561-83ec-f64c1bf17428)

**Reply by d5364t54ytfr4** [reply]
I don't know how the mental gymnastics for design choices of making it easy to show which tab is which container and which workspace can be done.
I useed to use `Container Sidebar` addon in native FF.

**Comment by fskaeh** [comment]
+1 on this feature request. Like most other people in this thread, I'm coming from Arc, and the way they do profile management over there is really great. Especially tying workspaces to profiles, allowing me to keep everything in one window and knowing that switching to my "work" workspace means I'm also switching to my work profile without extra manual intervention on my part. It's all very seamless and convenient.  
I know multi-account containers exist on Firefox, and I dabbled with them some, but they are frankly restrictive and cumbersome to use.

**Comment by mr-cheffy** [comment]
Workspaces now have builtin containers, thanks!

**Reply by digiaust** [reply]
This is really handy, but the next step would be the ability to link containers or workspaces to profiles. 

As @mauro-balades describes, Arc (who after the recent updates on basically the death of the project, I'd imagine you will have a TON of users start to migrate (look at google trends, you will see a big boost in zen adoption)) Arc has the ability to tie workspaces and containers to a specific profile. This is SUPER handy to be able to seamlessly switch between accounts in the same window instead of opening a separate window. (I have dabbled in firefox and know that their profile implementation is absolute donkey sh*t so I know this isn't an easy ask)

But the ability to seamlessly switch between profiles with different passwords, bookmarks, history, etc (like in arc) I think would be a HUGE feature for new salty users hurt by Arc to come to. I know this would make my life 100x easier and make Zen my default for life.

![image](https://github.com/user-attachments/assets/51b59280-a249-4b0b-8afa-24eb230eb5ad)
This keeps browser data separate, and gives access to all profile settings inside one window with a simple switch.
This would be huge.

**Reply by eracheee** [reply]
+1 This is a must.

**Comment by jeanlucthumm** [comment]
What about separating extensions? That way I can have a separate instance of Bitwarden for my Work workspace, which works on Arc

**Reply by digiaust** [reply]
Workspaces have their own "containers" now which is a step in the right direction but it only separates out history, cookies, and a few other things. Everything else, passwords, bookmarks, pins, etc is all still profile based. You using profiles still keeps everything as separate as two different browsers would, but unfortunately they will be in separate windows... probably for a while as a change at this level is more likely a firefox issue and not something zen could reasonably implement.

**Comment by kamalhm** [comment]
Hey folks, not sure on what's the current status of this. I tired switching profile but it created a new window. Did I do something wrong or we can't switch profile in one window yet?

**Comment by tornenen** [comment]
That's the only thing whats holding me off from switching from arc to zen. Have my own profile per Workspace. Own Essentials etc. per Workspace.

**Reply by vivianlauerbsu** [reply]
You should try it again because with container specific essentials it's basically already there

**Reply by tornenen** [reply]
Not really. 
I have the same essentials and the same bookmarks on every workspace.

**Reply by vraravam** [reply]
<img width="1548" alt="Screenshot 2025-01-07 at 07 20 49" src="https://github.com/user-attachments/assets/0d5a0331-928c-4640-8f84-081f22680907" />
in the Zen settings view, search for "container-specific" and you will find it.

**Reply by tornenen** [reply]
Oh, thats great. Thanks!
Then im a step further.
Now i only need different bookmarks and different addins per container.
or are there also options for this?

**Reply by thestrike72** [reply]
You can make different bookmarks per workspace. When you make a bookmark just use the workspace dropdown. Different extensions is not implemented. What is your use case for that?

**Reply by tornenen** [reply]
Great, now its 95% complete. 

For my private passwords i use bitwarden as pw manager, but my company is using keeper. So i need keeper in my work workspace and bitwarden in my private workspace.

**Reply by k-r-a-s-s** [reply]
this is really exciting. I agree 95% of the way, like @tornenen I also have a need for containing extensions - password managers, AI tools that are logged in to company specific accounts.

**Reply by digiaust** [reply]
It would be SO much more helpful if workspaces could be tied to a specific Firefox profile in about:profiles. 

This would take care of completely separating instances in workspaces. It would be optional of course for users who don't need separation, but if you look at why Arc was so popular, it was because it had unmatched workspace management including the ability to separate workspaces completely based on Profiles. This separated everything mentioned here in this request. Bookmarks, extensions, cookies/site data, history, themes, pinned tabs and for zen specifically essentials, active tabs, browser settings, tabs from other devices, 

I have limited understanding about development, but I know something like this would completely re-work the way zen manages profiles, windows, and workspaces, but in the future, I think I speak for a majority of users in this thread, that having profile-specific workspaces would be a God send and make Zen the go-to defacto browser for all the refugees from Arc.

**Comment by d5364t54ytfr4** [comment]
The current workaround is to use the official Firefox addon `Multi-Account Containers`.
This means the cookies and login data is isolated, but history and addons are shared

**Comment by mehmetkursataydin** [comment]
Can't wait for the feature to come too, please <3

**Comment by alrcatraz** [comment]
if we can use different profiles in different workspaces without opening multiple windows or restart Zen, Zen will be my favourite browser

**Comment by Adiiieee** [comment]
as soon as this feature comes I'm switching to zen. ( commenting just to get an update on this feature )

**Reply by thestrike72** [reply]
This already exists, just use the multi-account containers extension and set a container for each workspace. You can even have a separate set of essentials per container.

**Reply by tornenen** [reply]
But not different addons per Space. Or am i wrong?

This is the Feature iam missing.

**Reply by fskaeh** [reply]
> This already exists, just use the multi-account containers extension and set a container for each workspace. You can even have a separate set of essentials per container.

Doesn't fit the use case, this has been explained many times in this thread already.

**Reply by tornenen** [reply]
i think so too.
We need different profiles per workspace. The profile switch right now opens a different browser window.

**Comment by SaidBahaDev** [comment]
This is the most requested feature.

**Comment by DavidKaub** [comment]
Why is this issue closed?

**Comment by justbearcause** [comment]
There're 2 issues with Zen Browser that stopping me from switching to it completely — a lack of tab folders and this one. I think this discussion should be reopened.

**Comment by tornenen** [comment]
i guess this is something the dev wont implement. 
unlucky... i hope someone else will do it like arc. until then i need to stay with arc.

**Comment by Mheetu** [comment]
There's another post for this, be sure to upvote it!
- #2337  

The original post here was worded significantly better, but there's some good discussion on the other post.

**Comment by georgekatsanos-nc** [comment]
I'll need to move to a different browser if this doesn't get implemented.

**Reply by Malagutte** [reply]
You can have different work spaces, each one with with one container-profile associated with it.

Like you have the same history but at least you have different "profiles"
<img width="254" height="297" alt="image" src="https://github.com/user-attachments/assets/40553c15-aa11-4484-bccc-8220dce40b83" />

**Reply by fskaeh** [reply]
@Malagutte please read the rest of this discussion. This has been explained ad nauseam. Containers don't fulfill the requirements of a full browser profile.

**Comment by EnzoPV25** [comment]
Hi, I'm really interested on this feature. My use case is the following : 

I'm using different password manager between professional environment and personnal usage.

I need to be able to change the used extension per workspace to use keepass on professional workspace and bitwarden on personal workspace. 

Is this planned ?

**Comment by standerksen** [comment]
Would still love to see different profiles (not containers) per workspace. I don't get why this idea is closed.

**Comment by abiezerm** [comment]
I just create a [PR](https://github.com/zen-browser/desktop/pull/13709) trying to help this awsome project and community to have this feature, Im a newbe in firefox so lets see how it goes.

**Comment by vict0rsch** [comment]
Yet another push for this feature :)

I love Zen, I'm grateful for the work put in it. Containers are what has enabled me to make Zen usable but I still have pain points shared in this thread:

* Shared addons (I, too, have an issue with Password Managers. What's worse is I use the same provider so I can't just live with 2 different add-ons and choose manually...)
* Shared history
* Shared "Go to tab" from the command-palette which often makes me switch workspace involuntarily
* The media player is shared too 

Thanks!

**Comment by eracheee** [comment]
+1

Containers do not replace browser profiles. They solve a different problem.

This feature is really missing.

A Workspace should be able to use a dedicated Firefox Profile while staying in the same browser window.

For example:

* Personal Workspace → Personal Profile
* Work Workspace → Work Profile

Switching Workspace would automatically switch the associated profile.

This would allow each Workspace to have its own:

* extensions (e.g. Bitwarden Personal vs. Keeper Enterprise);
* download folder (e.g. `~/Downloads` vs. `~/pCloud/Work`);
* browsing history and address bar suggestions;
* browser settings and permissions.

Containers would still be useful for managing multiple accounts within the same profile.

❌ This is the one feature still preventing me from using Zen. I go back to Arc.

**Reply by vraravam** [reply]
If, as you describe, a workspace is equivalent to a Profile, then why use containers at all?

Looking at it the other way, I think what you are asking for is, while opening a link, if that link is "reigstered" to a specific Profile, then that profile window should open that url in a tab? Would that suffice for your usage? (just trying to understand your usecase)

**Reply by alrcatraz** [reply]
In my experience, switching between Profiles needs multiple windows. If such feature can be supported, we will be able to switch by switching containers in solely one window, which provides a much better user experience

**Comment by danulqua** [comment]
Upvoting the feature to have Spaces with different profiles, **not containers** – this is the only thing that stops me from migrating from Arc Browser, such a killer feature.
